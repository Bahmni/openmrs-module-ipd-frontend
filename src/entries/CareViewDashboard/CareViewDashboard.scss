.care-view-page {
  color: #393939;
  font-family: "IBM Plex Sans", sans-serif;
  .main {
    background-color: white;
    transition: margin-left 0.5s;
    width: 100%;
    top: 48px;
    left: 0;
    z-index: 1000;
    position: absolute;
    height: 100%;
    padding: 16px 0;
  }

  .home {
    background-color: #004144;
    fill: #ffffff;
    outline: none;
    border: none;
    margin-left: 16px;
  }

  .border-bottom-0 {
    border-bottom: 0;
  }

  .header-bg-color {
    background: #004144;
  }

  .header-nav-toggle-btn {
    border: none;
  }

  .header-nav-toggle-btn:hover {
    background-color: #004144;
    fill: #ffffff;
    outline: none;
    border: none;
  }
  .care-view-navigations {
    display: flex;
    flex-flow: row;
    justify-content: space-between;
    align-items: baseline;
    color: rgb(15, 98, 254);
    padding: 0 16px;
  }

  .ward-view-nav-link {
    display: flex;
    align-items: center;

    svg {
      margin-right: 8px;
    }
  }
}
