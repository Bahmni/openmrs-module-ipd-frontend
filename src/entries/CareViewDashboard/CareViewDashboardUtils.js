import { homePageUrl } from "../../constants";

export const goToHome = () => {
  window.location.href = homePageUrl;
};
