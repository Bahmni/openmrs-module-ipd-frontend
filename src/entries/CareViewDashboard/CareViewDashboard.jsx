import React from "react";
import PropTypes from "prop-types";
import { Header, Link } from "carbon-components-react";
import { Home20, ListBulleted24 } from "@carbon/icons-react";
import { ArrowLeft } from "@carbon/icons-react/next";
import "./CareViewDashboard.scss";
import { CareViewSummary } from "../../features/CareViewSummary/components/CareViewSummary";
import { CareViewPatients } from "../../features/CareViewPatients/components/CareViewPatients";
import { goToHome } from "./CareViewDashboardUtils";
import { FormattedMessage } from "react-intl";
import { I18nProvider } from "../../features/i18n/I18nProvider";

const CareViewDashboard = (props) => {
  const { hostData, hostApi } = props;
  console.log("hostData", hostData);
  console.log("hostApi", hostApi);
  return (
    <I18nProvider>
      <main className="care-view-page">
        <Header
          className="border-bottom-0 header-bg-color"
          aria-label="IBM Platform Name"
        >
          <Link onClick={() => goToHome()}>
            <Home20 className="home" aria-label="home-button" />
          </Link>
        </Header>

        <section className="main">
          <div className="care-view-navigations">
            <ArrowLeft
              data-testid={"Back button"}
              size={20}
              onClick={() => hostApi?.onHome()}
            />
            <Link
              className="ward-view-nav-link"
              onClick={() => hostApi?.onHome()}
            >
              <ListBulleted24 data-testid={"List button"} size={20} />
              <FormattedMessage
                id={"WARD_LIST_VIEW_TEXT"}
                defaultMessage="Ward List View"
              />
            </Link>
          </div>
          <CareViewSummary />
          <CareViewPatients />
        </section>
      </main>
    </I18nProvider>
  );
};

export default CareViewDashboard;

CareViewDashboard.propTypes = {
  hostData: PropTypes.object.isRequired,
  hostApi: PropTypes.object.isRequired,
};
