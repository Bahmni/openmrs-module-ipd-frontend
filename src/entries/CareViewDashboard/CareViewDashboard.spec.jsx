import React from "react";
import { render, waitFor } from "@testing-library/react";
import CareViewDashboard from "./CareViewDashboard";

describe("CareViewDashboard", () => {
  it("should match the snapshot", async () => {
    const { container, getByLabelText } = render(<CareViewDashboard />);
    await waitFor(() => {
      expect(getByLabelText("home-button")).toBeTruthy();
    });
    expect(container).toMatchSnapshot();
  });

  it("should render the component", async () => {
    const { getByLabelText } = render(<CareViewDashboard />);
    await waitFor(() => {
      expect(getByLabelText("home-button")).toBeTruthy();
    });
  });
});
