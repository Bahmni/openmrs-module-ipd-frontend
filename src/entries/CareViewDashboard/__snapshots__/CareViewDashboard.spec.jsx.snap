// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CareViewDashboard should match the snapshot 1`] = `
<div>
  <main
    class="care-view-page"
  >
    <header
      aria-label="IBM Platform Name"
      class="bx--header border-bottom-0 header-bg-color"
    >
      <a
        class="bx--link"
      >
        <svg
          aria-label="home-button"
          class="home"
          fill="currentColor"
          focusable="false"
          height="20"
          preserveAspectRatio="xMidYMid meet"
          role="img"
          viewBox="0 0 32 32"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.6123,2.2138a1.01,1.01,0,0,0-1.2427,0L1,13.4194l1.2427,1.5717L4,13.6209V26a2.0041,2.0041,0,0,0,2,2H26a2.0037,2.0037,0,0,0,2-2V13.63L29.7573,15,31,13.4282ZM18,26H14V18h4Zm2,0V18a2.0023,2.0023,0,0,0-2-2H14a2.002,2.002,0,0,0-2,2v8H6V12.0615l10-7.79,10,7.8005V26Z"
          />
        </svg>
      </a>
    </header>
    <section
      class="main"
    >
      <div
        class="care-view-navigations"
      >
        <svg
          aria-hidden="true"
          data-testid="Back button"
          fill="currentColor"
          focusable="false"
          height="20"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 32 32"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M14 26L15.41 24.59 7.83 17 28 17 28 15 7.83 15 15.41 7.41 14 6 4 16 14 26z"
          />
        </svg>
        <a
          class="bx--link ward-view-nav-link"
        >
          <svg
            aria-hidden="true"
            data-testid="List button"
            fill="currentColor"
            focusable="false"
            height="24"
            preserveAspectRatio="xMidYMid meet"
            size="20"
            viewBox="0 0 32 32"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="7"
              cy="9"
              r="3"
            />
            <circle
              cx="7"
              cy="23"
              r="3"
            />
            <path
              d="M16 22H30V24H16zM16 8H30V10H16z"
            />
          </svg>
          Ward List View
        </a>
      </div>
      <div
        class="care-view-summary"
      >
        <div
          class="bx--dropdown__wrapper bx--list-box__wrapper"
        >
          
          <div
            class="bx--dropdown bx--list-box"
            id="default"
          >
            <button
              aria-disabled="false"
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="downshift-0-label downshift-0-toggle-button"
              class="bx--list-box__field"
              id="downshift-0-toggle-button"
              title="General Ward (A)"
              type="button"
            >
              <span
                class="bx--list-box__label"
              >
                General Ward (A)
              </span>
              <div
                class="bx--list-box__menu-icon"
              >
                <svg
                  aria-label="Open menu"
                  fill="currentColor"
                  focusable="false"
                  height="16"
                  name="chevron--down"
                  preserveAspectRatio="xMidYMid meet"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                  />
                  <title>
                    Open menu
                  </title>
                </svg>
              </div>
            </button>
            <div
              aria-labelledby="downshift-0-label"
              class="bx--list-box__menu"
              id="downshift-0-menu"
              role="listbox"
              tabindex="-1"
            />
          </div>
        </div>
        <div
          class="summary-tiles"
        >
          <div
            class="bx--tile summary-tile"
          >
            Total patients
          </div>
          <div
            class="bx--tile summary-tile"
          >
            My patients
          </div>
          <div
            class="bx--tile summary-tile"
          >
            New patients
          </div>
          <div
            class="bx--tile summary-tile"
          >
            Scheduled for OT
          </div>
          <div
            class="bx--tile summary-tile"
          >
            To be discharged
          </div>
        </div>
      </div>
      <div
        class="care-view-patients-container"
      >
        <div
          class="care-view-patients"
        >
          <div
            class="task-type"
          >
            <div
              class="bx--dropdown__wrapper bx--list-box__wrapper"
            >
              
              <div
                class="bx--dropdown bx--list-box"
                id="default"
              >
                <button
                  aria-disabled="false"
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby="downshift-1-label downshift-1-toggle-button"
                  class="bx--list-box__field"
                  id="downshift-1-toggle-button"
                  title="All tasks"
                  type="button"
                >
                  <span
                    class="bx--list-box__label"
                  >
                    All tasks
                  </span>
                  <div
                    class="bx--list-box__menu-icon"
                  >
                    <svg
                      aria-label="Open menu"
                      fill="currentColor"
                      focusable="false"
                      height="16"
                      name="chevron--down"
                      preserveAspectRatio="xMidYMid meet"
                      role="img"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                      />
                      <title>
                        Open menu
                      </title>
                    </svg>
                  </div>
                </button>
                <div
                  aria-labelledby="downshift-1-label"
                  class="bx--list-box__menu"
                  id="downshift-1-menu"
                  role="listbox"
                  tabindex="-1"
                />
              </div>
            </div>
          </div>
          <ul
            class="bx--accordion patient-section bx--accordion--end bx--accordion--md"
          >
            <li
              class="bx--accordion__item"
            >
              <button
                aria-controls="accordion-item-1"
                aria-expanded="false"
                class="bx--accordion__heading"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="bx--accordion__arrow"
                  fill="currentColor"
                  focusable="false"
                  height="16"
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11 8L6 13 5.3 12.3 9.6 8 5.3 3.7 6 3z"
                  />
                </svg>
                <div
                  class="bx--accordion__title"
                  dir="auto"
                >
                  Patient 1
                </div>
              </button>
              <div
                class="bx--accordion__content"
                id="accordion-item-1"
              >
                Patient 1
              </div>
            </li>
            <li
              class="bx--accordion__item"
            >
              <button
                aria-controls="accordion-item-2"
                aria-expanded="false"
                class="bx--accordion__heading"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="bx--accordion__arrow"
                  fill="currentColor"
                  focusable="false"
                  height="16"
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11 8L6 13 5.3 12.3 9.6 8 5.3 3.7 6 3z"
                  />
                </svg>
                <div
                  class="bx--accordion__title"
                  dir="auto"
                >
                  Patient 2
                </div>
              </button>
              <div
                class="bx--accordion__content"
                id="accordion-item-2"
              >
                Patient 2
              </div>
            </li>
          </ul>
        </div>
      </div>
    </section>
  </main>
</div>
`;
